src/client.js

```diff
@@ -23,12 +23,29 @@ const API_TYPE_CLIENT = 'client';
 const API_TYPE_APP = 'app';
 
 /**
- * Create a new StitchClient instance.
+  * Factory class to create a new StitchClient asynchronously.
+  */
+export class StitchClient {
+  constructor() {
+    throw new StitchError('Constructor undefined. Use `StitchClient.init` instead.');
+  }
+
+  static init(clientAppID, options = {}) {
+    return new Promise((resolve, reject) => {
+      resolve(new _StitchClient(clientAppID, options));
```

Comment:
I don't see how this will actually wait for auth to initialize since this is still calling a constructor